### PR TITLE
You can now deconstruct AI cores that contain no AI

### LINKED ISSUE #20823: Unfastening/Fastening the AI core with a wrench shows reversed text.
image

As shown, Fastening an AI core with a wrench displays "Unfastening Core..." while Unfastening shows "Fastening core..."

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
rscadd: You can now deconstruct AI cores that contain no actual AI.
/:cl:

Fixes #20823

### CODE REVIEW

**Comment by lzimann** [comment]
the future... IS HERE

**Comment by KorPhaeron** [comment]
phil235: Joan: will conflict with mine

File: code/game/objects/structures/ai_core.dm [hunk 1]
```diff
@@ -25,218 +31,191 @@
...
 				return
-			if(istype(P, /obj/item/stack/cable_coil))
-				var/obj/item/stack/cable_coil/C = P
-				if(C.get_amount() >= 5)
+			playsound(loc, 'sound/items/Welder.ogg', 50, 1)
+			user << "<span class='notice'>You start to deconstruct the frame...</span>"
+			if(do_after(user, 20/P.toolspeed, target = src) && state != EMPTY_CORE && src && WT && WT.remove_fuel(0, user))
+				user << "<span class='notice'>You deconstruct the frame.</span>"
+				new /obj/item/stack/sheet/plasteel( loc, 4)
+				qdel(src)
```
Comment (phil235) [inline code/game/objects/structures/ai_core.dm]: call `deconstruct(TRUE) instead.`

File: code/game/objects/structures/ai_core.dm [hunk 2]
```diff
@@ -25,218 +31,191 @@
...
-					 "<span class='notice'>You start to [anchored ? "fasten [src] to" : "unfasten [src] from"] the floor...</span>")
-		switch(anchored)
-			if(0)
-				if(do_after(user, 20, target = src))
-					user << "<span class='notice'>You fasten the core into place.</span>"
-					anchored = 1
-			if(1)
-				if(do_after(user, 20, target = src))
-					user << "<span class='notice'>You unfasten the core.</span>"
-					anchored = 0
```
Comment (phil235) [inline code/game/objects/structures/ai_core.dm]: Hey, I know it's not your code, but `atom/proc/transfer_ai(` below is missing a `/` at the beginning.

File: code/game/objects/structures/ai_core.dm [hunk 3]
```diff
@@ -25,218 +31,190 @@
...
+			var/obj/item/weapon/weldingtool/WT = P
+			if(!WT.isOn())
+				user << "<span class='warning'>The welder must be on for this task!</span>"
 				return
-			if(istype(P, /obj/item/stack/cable_coil))
-				var/obj/item/stack/cable_coil/C = P
-				if(C.get_amount() >= 5)
+			playsound(loc, 'sound/items/Welder.ogg', 50, 1)
+			user << "<span class='notice'>You start to deconstruct the frame...</span>"
+			if(do_after(user, 20/P.toolspeed, target = src) && src && state == EMPTY_CORE && WT && WT.remove_fuel(0, user))
```
Comment (phil235) [inline code/game/objects/structures/ai_core.dm]: the src check is useless, what you need is !qdeleted(src) (or if you're cheap and feel lucky "if(loc)" )

**Pull request merged**